oak_runtime/src/lib.rs

```diff
@@ -636,6 +656,88 @@ impl Runtime {
         with_writer_channel(channel_half, |channel| Ok(channel.label.clone()))
     }
 
+    /// Returns the [`Label`] associated with the channel handle serialized as a byte array.
+    ///
+    /// If the serialized size is larger than the specified capacity, it will return a status
+    /// indicating the required capacity.
+    fn get_serialized_channel_label(
+        &self,
+        node_id: NodeId,
+        handle: oak_abi::Handle,
+        capacity: usize,
+    ) -> Result<LabelSerializationStatus, OakStatus> {
+        let half = self.abi_to_half(node_id, handle)?;
+        let label = match half.direction {
+            ChannelHalfDirection::Read => self.get_reader_channel_label(&half)?,
+            ChannelHalfDirection::Write => self.get_writer_channel_label(&half)?,
+        };
+
+        let size = label.encoded_len();
+        if size > capacity {
+            Ok(LabelSerializationStatus::NeedsCapacity(size))
+        } else {
+            let mut encoded = Vec::with_capacity(size);
+            match label.encode(&mut encoded) {
+                Err(error) => {
+                    warn!("Could not encode label: {}", error);
+                    Err(OakStatus::ErrInternal)
+                }
+                Ok(()) => Ok(LabelSerializationStatus::Success(encoded)),
+            }
+        }
+    }
+
+    /// Returns the [`Label`] associated with the node serialized as a byte array.
+    ///
+    /// If the serialized size is larger than the specified capacity, it will return a status
+    /// indicating the required capacity.
+    fn get_serialized_node_label(
+        &self,
+        node_id: NodeId,
+        capacity: usize,
+    ) -> Result<LabelSerializationStatus, OakStatus> {
+        let label = self.get_node_label(node_id);
+        let size = label.encoded_len();
+        if size > capacity {
+            Ok(LabelSerializationStatus::NeedsCapacity(size))
+        } else {
+            let mut encoded = Vec::with_capacity(size);
+            match label.encode(&mut encoded) {
+                Err(error) => {
+                    warn!("Could not encode label: {}", error);
+                    Err(OakStatus::ErrInternal)
+                }
+                Ok(()) => Ok(LabelSerializationStatus::Success(encoded)),
+            }
+        }
```

Comment:
A helper function (signature `(Label) -> Result<LabelSerializationStatus, OakStatus>`) would reduce the copy-pasta.